rust/chains/hyperlane-cosmos/src/interchain_gas.rs

```diff
@@ -0,0 +1,46 @@
+use async_trait::async_trait;
+use hyperlane_core::{
+    ChainResult, HyperlaneChain, HyperlaneContract, Indexer, InterchainGasPaymaster,
+};
+use hyperlane_core::{HyperlaneDomain, HyperlaneProvider, InterchainGasPayment, LogMeta, H256};
+
+/// A reference to a InterchainGasPaymaster contract on some Cosmos chain
+#[derive(Debug)]
+pub struct CosmosInterchainGasPaymaster {}
+
+impl HyperlaneContract for CosmosInterchainGasPaymaster {
+    fn address(&self) -> H256 {
```

Comment:
Cosmos currently has 43 / 63 length of the address, [and will be changed sometimes in future](https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-028-public-key-addresses.md#basic-address). So, how can we infer the original length of the address now? Should we just have to decide with whether this address has 12 bytes (24 length of zeros) prefix?

Reply:
Ah, afaik most CosmWasm contract address has full length of address. So this could not be the critical problem

Reply:
for ethereum we do a similar left padding of zeroes to convert between the 20 byte and 32 byte types
H256 and H160 have the `from` and `into` traits implemented